hopsworks-api/src/main/java/io/hops/hopsworks/api/dataset/DatasetResource.java

```diff
@@ -0,0 +1,386 @@
+    Users user = jwtHelper.getUserPrincipal(sc);
+    ResourceRequest resourceRequest = new ResourceRequest(ResourceRequest.Name.INODES);
+    Project project = this.getProject();
+    DatasetPath datasetPath = datasetHelper.getDatasetPathIfFileExist(project, path, datasetType);
+    InodeDTO dto;
+    switch (action == null? DatasetActions.Get.STAT : action) {
+      case BLOB:
+        dto = inodeBuilder.buildBlob(uriInfo, resourceRequest, project, user, datasetPath, mode);
+        break;
+      case LISTING:
+        resourceRequest.setOffset(pagination.getOffset());
+        resourceRequest.setLimit(pagination.getLimit());
+        resourceRequest.setSort(inodeBeanParam.getSortBySet());
+        resourceRequest.setFilter(inodeBeanParam.getFilter());
+        dto = inodeBuilder.buildItems(uriInfo, resourceRequest, project, datasetPath);
+        break;
+      case STAT:
+        if (datasetPath.isTopLevelDataset()) {
+          ResourceRequest datasetResourceRequest = new ResourceRequest(ResourceRequest.Name.DATASETS);
+          datasetResourceRequest.setExpansions(datasetExpansionBeanParam.getResources());
+          DatasetDTO datasetDTO = datasetBuilder.build(uriInfo, datasetResourceRequest, project,
+            datasetPath.getDataset(), null, null);
+          return Response.ok().entity(datasetDTO).build();
+        } else {
+          dto = inodeBuilder.buildStat(uriInfo, resourceRequest, datasetPath);
+        }
+        break;
+      default:
+        throw new WebApplicationException("Action not valid.", Response.Status.NOT_FOUND);
+    }
+    return Response.ok().entity(dto).build();
+  }
+  
+  @POST
+  @Path("{path: .+}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Post an action on a file.")
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  @JWTRequired(acceptedTokens = {Audience.API}, allowedUserRoles = {"HOPS_ADMIN", "HOPS_USER"})
+  @ApiKeyRequired( acceptedScopes = {ApiScope.DATASET_CREATE}, allowedUserRoles = {"HOPS_ADMIN", "HOPS_USER"})
+  public Response postByPath(@PathParam("path") String path, @QueryParam("type") DatasetType datasetType,
+    @QueryParam("target_project") String targetProjectName, @QueryParam("action") DatasetActions.Post action,
+    @QueryParam("templateId") Integer templateId, @QueryParam("description") String description,
+    @QueryParam("searchable") Boolean searchable, @QueryParam("generate_readme") Boolean generateReadme,
+    @QueryParam("destination_path") String destPath, @QueryParam("destination_type") DatasetType destDatasetType,
+    @Context UriInfo uriInfo, @Context SecurityContext sc) throws DatasetException, ProjectException,
+    HopsSecurityException {
+    Users user = jwtHelper.getUserPrincipal(sc);
+    DatasetPath datasetPath;
+    DatasetPath distDatasetPath;
+    Project project = this.getProject();
+    RESTApiJsonResponse json = new RESTApiJsonResponse();
```

Comment:
@ErmiasG do we need to use it? I think it's redundant since the status code indicates the result of the operation. Shouldn't it be client creating these messages?